subgraphs/compound-forks/protocols/cream-finance/src/mapping.ts

```diff
@@ -261,12 +269,49 @@ export function handleAccrueInterest(event: AccrueInterest): void {
   let oracleContract = PriceOracle.bind(
     Address.fromString(protocol._priceOracle)
   );
+
+  // cream finance uses an oracle that returns price in eth rather than usd.
+  // however, our common logic expects the oracle returned price to be in usd.
+  // to work around, for cream finance specifically, we get eth price first,
+  // multiple the oracle price by eth price, then pass it down to common logic.
+  //
+  // to compute eth price, we derive it from usdt price in eth.
+  // let's say if we know 1 usdt = 0.001 eth, then we know 1 eth = 1000 usd.
+  // to get usdt price, we simply do 10^(36-6)/getUnderlyingPrice(crUSDT), in which 6 = decimals(USDT)
+  //
+  // you may ask, why USDT, not USDC or DAI?
+  // this is bc crUSDT was deployed the earliest on mainnet compared to crUSDC and crDAI.
+  let usdtPriceResult = oracleContract.try_getUnderlyingPrice(
```

Comment:
should this specfic logic be wrapped by an "if ethereum network" since that mkt only exists on ethereum?

Reply:
@dmelotik  how about me adding the crSTABLE address to config? so that it easily covers all network :)

Reply:
I'm slightly confused. Would you have one for each network and have logic to switch based on the network?